Porting/add-pod-file

```diff
@@ -0,0 +1,239 @@
+
+This is a program which I<may> be helpful when a committer has to add a new
+F<*.pod> file in the F<pod/> directory.
+
+=head2 Prerequisites
+
+This program assumes that committer has taken the following steps (in the
+order listed):
+
+=over 4
+
+=item 1 You have run F<make test_prep>.
+
+This is to guarantee that all files are properly positioned.
+
+=item 2 You have placed a well-formatted F<.pod> file into the F<pod/> directory.
+
+In the C<NAME> section of this file there is a single non-blank line which
+consists of a string in the format C<STUB - ABSTRACT>, where C<STUB> is the
+basename of the file without the C<.pod> suffix and C<ABSTRACT> is the short
+description of the file.  For example, a new file whose path is
+F<pod/perlphonypod.pod> must have a C<NAME> section like this:
+
+    =head1 NAME
+
+    perlphonypod - This is phony POD
+
+=back
+
+F<pod/*.pod> files need entries in multiple locations to keep F<make
+test_porting> happy.  This program automates the formulation of I<most> of
+those entries, but will need some assistance from the committer to work
+properly.  The committer will have to make a reasonable choice as to which
+section of F<pod/perl.pod> the new F<.pod> file should be listed under.
+The eligible sections are shown in the following table:
+
+    Command-Line Value      Section in pod/perl.pod
+
+         O               => 'Overview',
+         T               => 'Tutorials',
+         R               => 'Reference Manual',
+         I               => 'Internals and C Language Interface',
+         M               => 'Miscellaneous',
+         L               => 'Language-Specific',
+         P               => 'Platform-Specific',
+
+For a first pass, we'll put the new entry at the end of the C<^=head2> section
+specified by the committer with the single-initial provided for command-line
+switch C<section>.
+
+=head2 Testing this program
+
+=over 4
+
+=item 1 Create a well formatted F<.pod> file somewhere on your system.
+
+=item 2 Copy it into the source tree under F<pod>.
+
+=item 3 Call the program as in L</USAGE> above.
+
+=item 4 Call F<git diff> and examine results.
+
+=item 5 Run F<make test_porting>.
+
+=back
+
+=cut
+
+my @man_sections = (
+    O   => 'Overview',
+    T   => 'Tutorials',
+    R   => 'Reference Manual',
+    I   => 'Internals and C Language Interface',
+    M   => 'Miscellaneous',
+    L   => 'Language-Specific',
+    P   => 'Platform-Specific',
+);
+
+my @man_section_abbrevs = ();
+my $man_sections_str = '';
+for (my $i=0; $i<= $#man_sections; $i+=2) {
+    my $j = $i+1;
+    push @man_section_abbrevs, $man_sections[$i];
+    $man_sections_str .= "\t$man_sections[$i] => $man_sections[$j]\n";
+}
+my %man_sections_seen = map { $_ => 1 } @man_section_abbrevs;
+my $man_sections = { @man_sections };
+
+my ($stub, $abstract, $section, $verbose) = ('') x 4;
+GetOptions(
+    "stub=s"        => \$stub,
+    "abstract=s"    => \$abstract,
+    "section=s"     => \$section,
```

Comment:
Above, under "well-formatted" you suggest that the pod file being added must have a `NAME` section with a stub and abstract, but here you require them in the program.  This seems like more work for the human running this program, as well as an opportunity to create disagreement.  I think you should extract the stub and abstract from the source file, and you should eliminate these options.

Reply:
> Above, under "well-formatted" you suggest that the pod file being added must have a `NAME` section with a stub and abstract, but here you require them in the program. This seems like more work for the human running this program, as well as an opportunity to create disagreement. I think you should extract the stub and abstract from the source file, and you should eliminate these options.

Yes, that was on mental TODO list and is the main reason I characterized this as a *first* pass.  Do you know if we have any prior art for parsing of a F<.pod> file to abstract what I call the 'stub' and the abstract?

I'll look at your other suggestions over the next several days.

Thank you very much.
Jim Keenan

Reply:
I don't know of prior art here but it should be parsed the same way as manpage indexers do.

one, or, more, comma-separated, names - rest of paragraph is the abstract

Reply:
> Yes, that was on mental TODO list and is the main reason I characterized this as a first pass. Do you know if we have any prior art for parsing of a F<.pod> file to abstract what I call the 'stub' and the abstract?

Module::Metadata can be helpful here:
```
my $data = Module::Metadata->new_from_file($filename, collect_pod => 1, decode_pod => 1);
my $abstract = $data->pod("ABSTRACT") // die "No abstract";
```

Reply:
Comments below
```
/ \A 
  \s+ # should be \s*
  \S+ # should allow for multiple names separated by comma and space
  \s? # should be \s+ - space before and after hyphen is required to disambiguate from names with hyphens in them
  - 
  \s? # ditto above
  (.+?) \s* \z /x
```

Reply:
Here is the regex dzil uses: https://metacpan.org/source/RJBS/Dist-Zilla-6.017/lib/Dist/Zilla/Util.pm#L54